modules/xmpp/JingleSessionPC.js

```diff
@@ -1252,111 +873,45 @@ JingleSessionPC.prototype.notifyMySSRCUpdate = function (old_sdp, new_sdp) {
         logger.info("Sending source-add", add.tree());
-        this.connection.sendIQ(add,
-            function (res) {
-                logger.info('got add result', res);
-            },
-            function (err) {
-                logger.error('got add error', err);
-            }
-        );
+        this.connection.sendIQ(
+            add, null, this.newJingleErrorHandler(add), IQ_TIMEOUT);
     } else {
         logger.log('addition not necessary');
     }
 };
 
-JingleSessionPC.prototype.getStats = function (interval) {
-    var self = this;
-    var recv = {audio: 0, video: 0};
-    var lost = {audio: 0, video: 0};
-    var lastrecv = {audio: 0, video: 0};
-    var lastlost = {audio: 0, video: 0};
-    var loss = {audio: 0, video: 0};
-    var delta = {audio: 0, video: 0};
-    this.statsinterval = window.setInterval(function () {
-        if (self && self.peerconnection && self.peerconnection.getStats) {
-            self.peerconnection.getStats(function (stats) {
-                var results = stats.result();
-                // TODO: there are so much statistics you can get from this..
-                for (var i = 0; i < results.length; ++i) {
-                    if (results[i].type == 'ssrc') {
-                        var packetsrecv = results[i].stat('packetsReceived');
-                        var packetslost = results[i].stat('packetsLost');
-                        if (packetsrecv && packetslost) {
-                            packetsrecv = parseInt(packetsrecv, 10);
-                            packetslost = parseInt(packetslost, 10);
-
-                            if (results[i].stat('googFrameRateReceived')) {
-                                lastlost.video = lost.video;
-                                lastrecv.video = recv.video;
-                                recv.video = packetsrecv;
-                                lost.video = packetslost;
-                            } else {
-                                lastlost.audio = lost.audio;
-                                lastrecv.audio = recv.audio;
-                                recv.audio = packetsrecv;
-                                lost.audio = packetslost;
-                            }
-                        }
-                    }
-                }
-                delta.audio = recv.audio - lastrecv.audio;
-                delta.video = recv.video - lastrecv.video;
-                loss.audio = (delta.audio > 0) ? Math.ceil(100 * (lost.audio - lastlost.audio) / delta.audio) : 0;
-                loss.video = (delta.video > 0) ? Math.ceil(100 * (lost.video - lastlost.video) / delta.video) : 0;
-                $(document).trigger('packetloss.jingle', [self.sid, loss]);
-            });
+JingleSessionPC.prototype.newJingleErrorHandler = function(request, failureCb) {
+    return function (errResponse) {
+
+        var error = { };
+
+        // Get XMPP error code and condition(reason)
+        const errorElSel = $(errResponse).find('error');
```

Comment:
Babel takes care of this (via the babel-plugin-transform-es2015-block-scoping plugin which is part of the ecma2015 preset). There is a difference between "var" and "const" (with the latter we will get "compile"-time errors and so detect bugs earlier), so it is not just a matter of consistency.